### ISSUE TITLE
Trouble installing on EC2 Amazon Linux 2

### ISSUE THREAD

**Issue by Lunteer** [phpredis/phpredis#1520]
### Expected behaviour

Nice clean install (But clearly an outdated version)

### Actual behaviour
--> Finished Dependency Resolution
Error: Package: php-pecl-redis-2.2.8-1.el7.x86_64 (epel)
           Requires: php(zend-abi) = 20100525-64
           Installed: php-common-7.2.14-1.amzn2.0.1.x86_64 (@amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-5.4.16-43.amzn2.x86_64 (amzn2-core)
               php(zend-abi) = 20100525-64
           Available: php-common-5.4.16-43.amzn2.0.1.x86_64 (amzn2-core)
               php(zend-abi) = 20100525-64
           Available: php-common-5.4.16-43.amzn2.0.2.x86_64 (amzn2-core)
               php(zend-abi) = 20100525-64
           Available: php-common-5.4.16-43.amzn2.0.3.x86_64 (amzn2-core)
               php(zend-abi) = 20100525-64
           Available: php-common-5.4.16-43.amzn2.0.4.x86_64 (amzn2-core)
               php(zend-abi) = 20100525-64
           Available: php-common-5.4.16-45.amzn2.0.5.x86_64 (amzn2-core)
               php(zend-abi) = 20100525-64
           Available: php-common-5.4.16-45.amzn2.0.6.x86_64 (amzn2-core)
               php(zend-abi) = 20100525-64
           Available: php-common-7.2.0-3.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.0-3.amzn2.1.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.4-3.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.5-3.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.5-3.amzn2.0.2.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.8-3.amzn2.0.2.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.8-3.amzn2.0.3.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.11-1.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
           Available: php-common-7.2.13-1.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(zend-abi) = 20170718-64
Error: Package: php-pecl-redis-2.2.8-1.el7.x86_64 (epel)
           Requires: php(api) = 20100412-64
           Installed: php-common-7.2.14-1.amzn2.0.1.x86_64 (@amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-5.4.16-43.amzn2.x86_64 (amzn2-core)
               php(api) = 20100412-64
           Available: php-common-5.4.16-43.amzn2.0.1.x86_64 (amzn2-core)
               php(api) = 20100412-64
           Available: php-common-5.4.16-43.amzn2.0.2.x86_64 (amzn2-core)
               php(api) = 20100412-64
           Available: php-common-5.4.16-43.amzn2.0.3.x86_64 (amzn2-core)
               php(api) = 20100412-64
           Available: php-common-5.4.16-43.amzn2.0.4.x86_64 (amzn2-core)
               php(api) = 20100412-64
           Available: php-common-5.4.16-45.amzn2.0.5.x86_64 (amzn2-core)
               php(api) = 20100412-64
           Available: php-common-5.4.16-45.amzn2.0.6.x86_64 (amzn2-core)
               php(api) = 20100412-64
           Available: php-common-7.2.0-3.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.0-3.amzn2.1.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.4-3.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.5-3.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.5-3.amzn2.0.2.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.8-3.amzn2.0.2.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.8-3.amzn2.0.3.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.11-1.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
           Available: php-common-7.2.13-1.amzn2.0.1.x86_64 (amzn2extra-php7.2)
               php(api) = 20170718-64
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
### I'm seeing this behaviour on
- OS: Amazon Linux 2 (I believe a fork of Fedora
- Redis: None
- PHP: 7.2
- phpredis: none yet

### Steps to reproduce, backtrace or example script

yum install "php-pecl-redis"

I am not an experienced developer so please go easy. I find the installation guide also a bit hard to follow as I tried installing and ran into issue there also. I do prefer package installations but obviously this one is very outdated.

Any help would be appreciated.

**Comment by remicollet** [comment]
>  Error: Package: php-pecl-redis-2.2.8-1.el7.x86_64 (epel)

Packages in EPEL are designed to work with packages in base CentOS / RHEL repositories, so with PHP 5.4.

If your PHP provider (amzn) don't provides "redis" extension you have to build it yourself, or choose a better provider.

**Comment by remicollet** [comment]
BTW, this issue is related to RPM packages, and repository conflicts, nothing with redis extension, so can be close.

**Comment by Lunteer** [comment]
OK, Thanks will close. Will try to build it from source again. But if you can help a noob out with some issues I had first time round.

- Where should I unzip/put the git folder. In my home folder?
- How do I ensure I get updates (manually?)
- After running make && make install do I need to move files around?

Thanks

**Comment by remicollet** [comment]
simply:  ` pecl install redis`

**Comment by Lunteer** [comment]
Thanks. I got an error of `pecl: command not found`. So obviously need to install it. I ran `sudo yum install php-pear` and then `pecl install redis` and it now works! Thanks for the help.

Added note for anyone else visiting. You might need some other extensions like php-devel or gcc. You can add them to the install `sudo yum install php-pear php-devel gcc`

@remicollet  Thanks for all the help and patience!!

**Comment by Lunteer** [comment]
Just for future reference if anyone wants to know but it seems like the conflicts have been solved and now you can install using `yum install php-pecl-redis`
